journal/src/main/java/io/camunda/zeebe/journal/file/SegmentFileManager.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.journal.file;
+
+import io.camunda.zeebe.journal.JournalException;
+import io.camunda.zeebe.journal.file.record.CorruptedLogException;
+import io.camunda.zeebe.util.FileUtil;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileChannel.MapMode;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.OpenOption;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class SegmentFileManager {
```

Comment:
:wrench: From the functionality, it can: `loadExistingSegment` and `createSegment`. I think `SegmentLoader` is fine as a name, since tbh the `JournalSegments` is the one which "manages" segments (e.g. keeps track of them, handles corruption, deletion, etc.).

```suggestion
final class SegmentLoader {
```